File: assertion-roulette/csharp-xunit/OrganizationUnitsControllerTests_Should_Get_OrganizationUnit_Async.cs
using Xunit;

namespace BackOfficeBase.Tests.Web.Api
{
    public class OrganizationUnitsControllerTests
    {
        [Fact]
        public void Should_Get_OrganizationUnit_Async()
        {
            {
                var organizationUnitAppServiceMock = new Mock<IOrganizationUnitAppService>();
                organizationUnitAppServiceMock.Setup(x => x.GetAsync(It.IsAny<Guid>())).ReturnsAsync(new OrganizationUnitOutput { Id = Guid.NewGuid(), SelectedUsers = new[] { GetTestUserOutput("test_user_for_get_ou") }, SelectedRoles = new[] { GetTestRoleOutput("test_role_for_get_ou") } });
                var organizationUnitsController = new OrganizationUnitsController(organizationUnitAppServiceMock.Object);
                var actionResult = await organizationUnitsController.GetOrganizationUnits(Guid.NewGuid());
                var okObjectResult = Assert.IsType<OkObjectResult>(actionResult.Result);
                var organizationUnitOutput = Assert.IsType<OrganizationUnitOutput>(okObjectResult.Value);
                Assert.Equal((int)HttpStatusCode.OK, okObjectResult.StatusCode);
                Assert.True(organizationUnitOutput.SelectedUsers.Any());
                Assert.True(organizationUnitOutput.SelectedRoles.Any());
            }
        }
    }
}

File: assertion-roulette/python-unittest/OrganizationUnitsControllerTests_Should_Get_OrganizationUnit_Async.py
import unittest
from unittest.mock import Mock

class TestOrganizationUnitsController(unittest.TestCase):
    def test_should_get_organization_unit(self):
        organization_unit_app_service_mock = Mock()
        organization_unit_app_service_mock.get_async.return_value = OrganizationUnitOutput(
            id=str(uuid.uuid4()),
            selected_users=[get_test_user_output("test_user_for_get_ou")],
            selected_roles=[get_test_role_output("test_role_for_get_ou")]
        )
        
        organization_units_controller = OrganizationUnitsController(organization_unit_app_service_mock)
        action_result = organization_units_controller.get_organization_units(str(uuid.uuid4()))
        
        ok_object_result = self.assertIsInstance(action_result.result, OkObjectResult)
        organization_unit_output = self.assertIsInstance(ok_object_result.value, OrganizationUnitOutput)
        
        self.assertEqual(200, ok_object_result.status_code)
        self.assertTrue(len(organization_unit_output.selected_users) > 0)
        self.assertTrue(len(organization_unit_output.selected_roles) > 0)
